### PR TITLE
txscript: Rework check pubkey encoding test.

### PR DESCRIPTION
**Pull request by davecgh**
This reworks the tests for `CheckPubKeyEncoding` to test for the explicit reason for the failure to ensure that each test is actually testing the intended condition and be more consistent with the more recent tests.

It also adds several more tests to better ensure proper behavior and includes a couple of corner cases where public keys will ultimately fail validation due to not being on the curve, but are still valid encodings from the standpoint of preventing malleability.

### CODE REVIEW

File: txscript/consensus_test.go [hunk 1]
```diff
@@ -244,59 +245,98 @@ func TestCheckSignatureEncoding(t *testing.T) {
...
+			"ce0b14fb842b1ba549fdd675c98075f12e9c510f8ef52bd021a9a1f4809d3b4d"),
+		err: nil,
+	}, {
+		name: "compressed ok (ybit = 1)",
+		key: hexToBytes("03" +
+			"2689c7c2dab13309fb143e0e8fe396342521887e976690b6b47f5b2a4b7d448e"),
+		err: nil,
+	}, {
+		// Although public keys not on the curve will ultimately be invalid, it
+		// is considered a valid encoding from from the standpoint of preventing
```
Comment (rstaudt2) [inline txscript/consensus_test.go:275]: Typo here `from from` (and in the following comment as well)

Comment (davecgh) [reply txscript/consensus_test.go]: Nice catch.  Fixed.

**Comment by davecgh** [comment]
Rebased to resolve merge conflicts.
